All Checks have to work in defalt configuration without any Exceptions, issue #1129.

MEZk opened the pull request:
***Checks which are commented in checkstyle_checks.xml:***
1) Added UniquePropertiesCheck into checkstyle_checks.xml with default configuration. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheckTest.java#L60).
2) AvoidInlineConditionalsCheck works in default config without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/AvoidInlineConditionalsCheckTest.java#L33).
3) BooleanExpressionComplexity works in default config without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/metrics/BooleanExpressionComplexityCheckTest.java#L32).
4) Added [test with default configuration](https://github.com/MEZk/checkstyle/commit/b8eed0ca23209ee82ca612e52c0ab6a94a06478b#diff-505cbbd2e5af4b7accc76c91e12d1256R52) for ClassDataAbstractionCouplingCheck.
5) Added [test with default configuration](https://github.com/MEZk/checkstyle/commit/f67edba12c349ec9d7e0188e21de9c7409120755#diff-01360202649d4d131dbffbe3d639c344R59) for  ClassFanOutComplexityCheck.
6) Added [test with default configuration](https://github.com/MEZk/checkstyle/commit/c858b4703cd35e20da1cf778210dadf4ee4cf39b#diff-54f1f9ded82ca9b567f0f85e7957f59bR427) for  CustomImportOrderCheck.
7) DesignForExtensionCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheckTest.java#L33).
8) Added [test with default configuration](https://github.com/MEZk/checkstyle/commit/5f2fc9c2e5f59aef9c5ae8efe078ddef8c57d84d#diff-af11b6d030c4fa9884e6119be9bbb4cdR147) for  ExecutableStatementCountCheck.
9) FinalParametersCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheckTest.java#L30).
10) HiddenFieldCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheckTest.java#L62).
11) IllegalTokenCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenCheckTest.java#L31).
12) IllegalTypeCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheckTest.java#L42).
13) ImportOrderCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheckTest.java#L46).
14) InnerTypeLastCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/design/InnerTypeLastCheckTest.java#L34).
15) JavadocParagraph works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheckTest.java#L49).
16) JavadocTagContinuationIndentation works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTagContinuationIndentationCheckTest.java#L42).
17) Added [test with default configuration](https://github.com/MEZk/checkstyle/commit/11793f53ae97b302089f8e00bef5b2f1973a6db0#diff-267008c864da3b98ca536cb914331fccR64) for  JavaNCSSCheck.
18) MissingCtorCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingCtorCheckTest.java#L34).
19) ModifiedControlVariableCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/ModifiedControlVariableCheckTest.java#L33).
20) Added [test with default configuration](https://github.com/MEZk/checkstyle/commit/65ac9165014b9250eec386866fa7641cc7c4a47d#diff-3f28ccfc9f7fc957d8d665c777dca35eR109) for  MultipleStringLiteralsCheck.
21) NonEmptyAtclauseDescriptionCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheckTest.java#L33).
22) Added [test with default configuration](https://github.com/MEZk/checkstyle/commit/1fa57e840dd63d4734b1f6a72fe2671009254313#diff-af0c73cf17aca87f4a69141240affe78R72) for  NPathComplexityCheck.
23) OneTopLevelClassCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/design/OneTopLevelClassCheckTest.java#L99).
24) OverloadMethodsDeclarationOrderCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheckTest.java#L34).
25) PackageDeclarationCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheckTest.java#L34).
26) TrailingCommentCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheckTest.java#L38).
27) RequireThisCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java#L35).
28) ReturnCountCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/ReturnCountCheckTest.java#L31).
29) SingleLineJavadocCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SingleLineJavadocCheckTest.java#L31).
30) Added [test with default configuration](https://github.com/MEZk/checkstyle/commit/7f0edd77420fd805c15499b3754cc3fe120bcebb#diff-4ee920a468c047a2df4f0096396dc569R87) for  SummaryJavadocCheck.
31) Added [test with default configuration](https://github.com/checkstyle/checkstyle/commit/023e7e50d41e885d37318b3ff494a0c36caed5d6#diff-b0ea0ff8351325080c1ad76bc3035270R213) for  VariableDeclarationUsageDistanceCheck.
32) SingleLineJavadocCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheckTest.java#L53).
33) IllegalCatchCheck works in default configuration without problems. [Test for default configuration](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchCheckTest.java#L34).

***All checks which are specified in checkstyle_checks.xml were also checked with checkstyle-tester (checkstyle 6.8.1-RELEASE, sevntu-checkstyle) in default configuration and there were no exceptions.***
![screenshot-1](https://cloud.githubusercontent.com/assets/7242568/8801775/afe54fc6-2fc5-11e5-9b41-dd564d0c2d99.png)
![screenshot](https://cloud.githubusercontent.com/assets/7242568/8801776/afe66dde-2fc5-11e5-96b0-fc741627e39f.png)

[checkstyle-tester config file](http://mezk.github.io/my_check.xml)
[cobertura coverage report after all additions](http://mezk.github.io/cobertura_issue1129)

romani commented:
1)
> nested.if.depth

if message is duplicated, remove it, no need to comment it out.

2) 
> import com.puppycrawl.tools.checkstyle.checks.metrics.ClassFanOutComplexityCheck;

remove that from original commit, not need to make a rubbish in one commit and then remove at the last, do right from the beginning

3)
> CustomImportOrderCheckTest.java

What is a reason of introduced formatting changes ? do you do it manually or by Idea formatting ?
that is not good to mix such changes.

MEZk commented:
@romani 
1)
>if message is duplicated, remove it, no need to comment it out.

There was a mistake in property file. The second  ```nested.if.depth``` should be ```nested.try.depth```

2)
>remove that from original commit, not need to make a rubbish in one commit and then remove at the last, do right from the beginning

My mistake. Corrected.

3)
>What is a reason of introduced formatting changes ? do you do it manually or by Idea formatting ?
that is not good to mix such changes.

Sorry, it was due to IntelliJ IDEA formatting and my carelessness when I was committing. Corrected.

romani commented:
merged as FF.

thanks a lot for clear and detailed reports , they help a lot to speedup merge process.
